### ISSUE TITLE
no-invalid-regexp rule doesn't error with example from its documentation

### ISSUE THREAD

**Issue by FloEdelmann** [eslint/eslint#10861]
**Tell us about your environment**

* **ESLint Version:** 5.6.0
* **Node Version:** 8.12.0
* **npm Version:** 6.3.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "env": {
    "es6": true,
    "node": true
  },
  "parserOptions": {
    "ecmaVersion": 2017
  },
  "rules": {
    "no-invalid-regexp": "error"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// like in the no-invalid-regexp documentation:
// https://github.com/eslint/eslint/blob/master/docs/rules/no-invalid-regexp.md#rule-details
new RegExp('\\');
```

```bash
eslint file.js
```

**What did you expect to happen?**

eslint failing with an error from the `no-invalid-regexp` rule.

**What actually happened? Please include the actual, raw output from ESLint.**

eslint passed with no output.

**Comment by FloEdelmann** [comment]
It was still working with eslint 4.15.0. Could it be a regression introduced in #10062?

**Comment by sstern6** [comment]
Thanks @not-an-aardvark ill work on this.  Might have some questions.

**Comment by platinumazure** [comment]
Is this really a bug in the rule? I'm reading the example as a regex which matches against a backslash anywhere in the string.

My understanding is that escaping the backslash would be required in both the `RegExp` constructor and in a regexp literal:

```js
new RegExp("\\");

/\\/;
```

If I'm misunderstanding something, I would be grateful to learn what that is. Otherwise, I think this should be a documentation change (to remove the example), not a bugfix.

**Comment by not-an-aardvark** [comment]
When I run that code, I get an error:

```
SyntaxError: Invalid regular expression: /\/: \ at end of pattern
```

**Comment by not-an-aardvark** [comment]
[![](https://imgs.xkcd.com/comics/backslashes.png)](https://xkcd.com/1638/)

**Comment by sstern6** [comment]
@mysticatea im not understanding, how is it a valid pattern when `new RegExp("\\");` returns a syntax error?

**Comment by mysticatea** [comment]
I'm sorry, I misread it as `//` :sweat_smile:

**Comment by FloEdelmann** [comment]
`/\/` is an invalid RegExp, as the backslash is not escaped.  
`/\\/` is fine.  
In `new RegExp('\\')`, the backslash is just escaped in the string (in a string, e.g. `\n` is a valid character), so to escape it in both the string and the RegExp, one has to use `new RegExp('\\\\)`.

**Comment by mysticatea** [comment]
So this is a bug in [regexpp](https://www.npmjs.com/package/regexpp); it hasn't followed the https://github.com/tc39/ecma262/pull/910 spec update yet because I had overlooked it. I will update it.

**Comment by sstern6** [comment]
@mysticatea I was going to ask if this was a regexpp issue!  Thanks for the PR, so when this gets merged in we will need to update the regexpp package and then I was thinking that we should add a test to the test suite to avoid regressions for this use case?

Is this correct?

**Comment by not-an-aardvark** [comment]
@sstern6 It looks like the fix has already been published as a new version of `regexpp`. https://github.com/eslint/eslint/pull/10920 has the effect of upgrading `regexpp` and adding a test, as you suggested.

**Comment by FloEdelmann** [comment]
Thanks for fixing this!
